### PR TITLE
feat: readContractState in JS

### PR DESCRIPTION
**Pull request by andreespirela**

### CODE REVIEW

File: js/executor_test.ts [hunk 1]
```diff
@@ -1,24 +1,46 @@
...
 
-Deno.test("contract_load_test", async function () {
+localStorage.clear();
+
+Deno.test("contract_load_test", async () => {
   await loadContract("t9T7DIOGxx4VWXoCEeYYarFYeERTpWIC1V3y-BPZgKE");
 });
 
-Deno.test("execute_js_test", async function () {
+Deno.test("execute_js_test", async () => {
```
Comment (andreespirela) [inline js/executor_test.ts:10]: both `execute_js_test` are failing @littledivy

File: js/sw.js [hunk 2]
```diff
@@ -204,49 +396,76 @@ const WORKER = `{
...
 
   async resolveState() {
     this.#state = await new Promise((resolve) => {
-      this.#module.onmessage = (e) => {
-        resolve(e.data);
-        this.#module.terminate();
+      this.#module.onmessage = async (e) => {
+        if(e.data.readContractState) {
+          const { contractId, key, returnValidity, height, currentHeight } = e.data;
+          const { state, validity } = await this.executor.executeContract(contractId, height || currentHeight, true, this.gateway, returnValidity);
```
Comment (andreespirela) [inline js/sw.js:430]: If we pass a `false` value for the `clearCache` in the executeContract, it gets in an infinite loop. This is vital to take a look at why it does that when it has cached content before we merge this @littledivy

File: js/executor_test.ts [hunk 3]
```diff
@@ -1,24 +1,46 @@
...
+  let { state, validity } = await executeContract(
+      "Vjt13JlvOzaOs4St_Iy2jmanxa7dc-Z3pDk3ktwEQNA",
+      undefined,
+      true,
+      undefined,
+      true
+  );
+  console.log({ state, validity});
+
+  // console.log(state);
```
Comment (littledivy) [inline js/executor_test.ts:43]: Make an assertion.

**Review by littledivy — CHANGES_REQUESTED** [review]

File: js/executor.js [hunk 4]
```diff
@@ -1,355 +1,207 @@
...
+                // Slower than `rt.executeInteractions` but more readable
+                // 100 interactions in ~30.06ms.
+                //
+                // for (const interaction of interactions) {
+                //    const input = interaction.node.tags.find(data => data.name === "Input");
+                //    await rt.execute({ input, caller: interaction.node.owner.address });
+                // }
+
+                // Faster. At 100 interactions in about 3.68ms.
+                await rt.executeInteractions(interactions, tx);
```
Comment (andreespirela) [inline js/executor.js:143]: can we make sure `interactions` and `updatedInteractions` ABSOLUBTELY respect `height`, even if there's higher heights cached, we should have a filter `<= height`

Comment (andreespirela) [reply js/executor.js:143]: @littledivy

**Pull request merged**